snappy/build_test.go

```diff
@@ -29,11 +29,37 @@ import (
 	"strings"
 	"syscall"
 
+	"github.com/ubuntu-core/snappy/dirs"
 	"github.com/ubuntu-core/snappy/helpers"
+	"github.com/ubuntu-core/snappy/testutil"
 
 	. "gopkg.in/check.v1"
 )
 
+type BuildTestSuite struct {
+	testutil.BaseTest
+}
+
+var _ = Suite(&BuildTestSuite{})
+
+func (s *BuildTestSuite) SetUpTest(c *C) {
+	s.BaseTest.SetUpTest(c)
+
+	// chdir into a tempdir
+	pwd, err := os.Getwd()
+	c.Assert(err, IsNil)
+	s.AddCleanup(func() { os.Chdir(pwd) })
+	err = os.Chdir(c.MkDir())
+	c.Assert(err, IsNil)
+
+	// fake "du"
+	duCmd = makeFakeDuCommand(c)
+	s.AddCleanup(func() { duCmd = "du" })
```

Comment:
This is assuming it knows what duCmd originally was, which means this might potentially change the implementation unintendedly.

The pattern I often use in these cases is to have a function such as:
```
func fakeDuCmd(c *check.C) (reset func()) {
    origDuCmd := duCmd
    ...
    return func() { duCmd = origDuCmd }
}
```

and then use it as

```
reset := fakeDuCmd(c)
defer reset()
```

or `AddCleanup(reset)`, in your case.